source/RemoteWiring/HardwareProfile.h

```diff
@@ -0,0 +1,351 @@
+                {
+                    if( isDigitalOutputSupported( pin ) )
+                    {
+                        vector->Append( pin );
+                    }
+                }
+            }
+            return vector;
+        }
+    }
+
+    property Windows::Foundation::Collections::IVector<uint8_t> ^DisabledPins
+    {
+        Windows::Foundation::Collections::IVector<uint8_t> ^ get()
+        {
+            auto vector = ref new Platform::Collections::Vector<uint8_t>();
+            if( _is_valid )
+            {
+                for( uint8_t pin = 0; pin < _total_pins; ++pin )
+                {
+                    if( getPinCapabilitiesBitmask( pin ) == 0 )
+                    {
+                        vector->Append( pin );
+                    }
+                }
+            }
+            return vector;
+        }
+    }
+
+    property Windows::Foundation::Collections::IVector<uint8_t> ^I2cPins
+    {
+        Windows::Foundation::Collections::IVector<uint8_t> ^ get()
+        {
+            auto vector = ref new Platform::Collections::Vector<uint8_t>();
+            if( _is_valid )
+            {
+                for( uint8_t pin = 0; pin < _total_pins; ++pin )
+                {
+                    if( isI2cSupported( pin ) )
+                    {
+                        vector->Append( pin );
+                    }
+                }
+            }
+            return vector;
+        }
+    }
+
+    property Windows::Foundation::Collections::IVector<uint8_t> ^PwmPins
+    {
+        Windows::Foundation::Collections::IVector<uint8_t> ^ get()
+        {
+            auto vector = ref new Platform::Collections::Vector<uint8_t>();
+            if( _is_valid )
+            {
+                for( uint8_t pin = 0; pin < _total_pins; ++pin )
+                {
+                    if( isPwmSupported( pin ) )
+                    {
+                        vector->Append( pin );
+                    }
+                }
+            }
+            return vector;
+        }
+    }
+
+    property Windows::Foundation::Collections::IVector<uint8_t> ^ServoPins
+    {
+        Windows::Foundation::Collections::IVector<uint8_t> ^ get()
+        {
+            auto vector = ref new Platform::Collections::Vector<uint8_t>();
+            if( _is_valid )
+            {
+                for( uint8_t pin = 0; pin < _total_pins; ++pin )
+                {
+                    if( isServoSupported( pin ) )
+                    {
+                        vector->Append( pin );
+                    }
+                }
+            }
+            return vector;
+        }
+    }
+    
+    ///<summary>
+    ///This default constructor accepts an IBuffer containing pin information which is assumed to be in the default Firmata protocol.
+    ///<param name="buffer_">The input IBuffer object reference</param>
+    ///</summary>
+    [Windows::Foundation::Metadata::DefaultOverload]
```

Comment:
If there were an interface this should not be in it, but it would be implied in an implementation such as this (FirmataProfile).